src/tlo/methods/healthsystem.py

```diff
@@ -1725,7 +1725,7 @@ def write_to_hsi_log(
             },
             description="record of each HSI event"
         )
-        if did_run:
+        if did_run and event_details.appt_footprint:
```

Comment:
does this really check if the appt_footprint is blank (i.e. an empty dictionary, or a dictionary for which all values are 0)?

Reply:
It checks if it is an empty dictionary - or, least I think it does, in all of my tests. Would it also need to check if it is zero?

Reply:
this is something I'm not quire sure about. 
Is it the case that using the following three things to express a blank footprint from the HSI_Event are all equivalent?

* `return sim.modules['HealthSystem'].get_blank_footprint()`
* `return self.make_appt_footprint({})`
* `return {}`

If not, we need to make sure that the logic at this place interprets them as such.

I think the code as written is checking for 'truthiness' of the footprint, and a footprint that has all values of 0's (which we think of as being blank), would seem truth, and that therefore, this would not work as intended.

Reply:
Ah okay, I think that I was under the impression from last weeks meeting that all the zeroes were seen as equivalent, as we only wanted to track what actually had a non-zero footprint.

Reply:
the appt-footprint is always a dict. If all the values in that dict are zero, we say it's "blank".

But that's different to the python concept of it being 'falsey' versus 'truthy'.

So I think this check should be something like
```python
if did_run and sum(event_details.appt_footprint.values():
```
instead of 
```python
if did_run and event_details.appt_footprint
```

Reply:
Ah okay, understood

Reply:
So it looks like not all of the footprints are, in fact, dictionaries? Signalling a wider issue, or is it okay?

<img width="1080" alt="image" src="https://github.com/UCL/TLOmodel/assets/106470783/8b3e0299-7369-473e-8ace-c2010e72f6a7">

Reply:
are you able to find which HSI returned that, and what is returned?

Reply:
Yes, I think it's coming from run_individual_level_events_in_mode_0_or_1 in healthsystems.py... Seems to be the common thread connecting all of the test failures. 

Could then either convert the tuples to dictionaries when logging or add another check for tuples? 

<img width="1080" alt="image" src="https://github.com/UCL/TLOmodel/assets/106470783/84b40bd3-53ea-4223-b472-29c96c22f23e">

Reply:
I've had a look at this more closely, and I can see that the property being addressed here `event_details` is actually passed in as a NamedTuple (from [here](https://github.com/UCL/TLOmodel/blob/e9263803787825168955b0e24ed15c383a2ba0dd/src/tlo/methods/healthsystem.py#L1692-L1698)).

The conversion to NamedTuple is done [here](https://github.com/UCL/TLOmodel/blob/e9263803787825168955b0e24ed15c383a2ba0dd/src/tlo/methods/hsi_event.py#L406-L413).

Sorry to send you on a wild goose chase! 

This actually simplifies things, because I believe we can access all the values of NamedTuple using `list`. So therefore the check for "blankness" would be `sum(list(event_details.appt_footprint))`

Reply:
Ah okay, I was a bit behind you tracking it down. But I'll make that change now. 

Thank you for investigating!!